From 5e26b32a47dd65d4a8b5c8392b10a12e71fb038f Mon Sep 17 00:00:00 2001
From: Sudan Landge <quic_sudanvl@quicinc.com>
Date: Mon, 29 Aug 2022 18:34:03 +0530
Subject: [PATCH 1/1] ima: do not add any measure rules to improve boot time

All appraisal is purely based on digital signature for
sa515m and happens locally. We can save some boot time
by not generating formatted templates.

This commit removes rules that tells driver to generate
those templates.

Change-Id: I298f70ab81b18c3f034bba2b29cd8bda4dab8f2c
---
 security/integrity/ima/ima_policy.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/security/integrity/ima/ima_policy.c b/security/integrity/ima/ima_policy.c
index 8f90e87ec18e..87908afd0dbf 100644
--- a/security/integrity/ima/ima_policy.c
+++ b/security/integrity/ima/ima_policy.c
@@ -682,21 +682,6 @@ void __init ima_init_policy(void)
 		add_rules(dont_measure_rules, ARRAY_SIZE(dont_measure_rules),
 			  IMA_DEFAULT_POLICY);
 
-	switch (ima_policy) {
-	case ORIGINAL_TCB:
-		add_rules(original_measurement_rules,
-			  ARRAY_SIZE(original_measurement_rules),
-			  IMA_DEFAULT_POLICY);
-		break;
-	case DEFAULT_TCB:
-		add_rules(default_measurement_rules,
-			  ARRAY_SIZE(default_measurement_rules),
-			  IMA_DEFAULT_POLICY);
-		break;
-	default:
-		break;
-	}
-
 	/*
 	 * Based on runtime secure boot flags, insert arch specific measurement
 	 * and appraise rules requiring file signatures for both the initial
-- 
2.44.1

